internal/pkg/client/storageos/v2/client.go

```diff
@@ -0,0 +1,114 @@
+package v2
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	storageosapiv2 "github.com/storageos/go-api/v2"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/storageos/cluster-operator/internal/pkg/client/storageos/common"
+	storageosv1 "github.com/storageos/cluster-operator/pkg/apis/storageos/v1"
+)
+
+// ErrNoAuthToken is returned when the API client did not get an error
+// during authentication but no valid auth token was returned.
+var ErrNoAuthToken = errors.New("no token found in auth response")
+
+// NewClient creates a new StorageOS v2 client and returns it.
+func NewClient(ip, username, password string) (context.Context, *storageosapiv2.APIClient, error) {
+	config := storageosapiv2.NewConfiguration()
+
+	u, err := url.Parse(fmt.Sprintf("%s://%s:%d", common.DefaultScheme, ip, storageosv1.DefaultServiceExternalPort))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	config.Scheme = u.Scheme
+	config.Host = u.Host
+	config.UserAgent = common.UserAgent
+
+	httpc := &http.Client{
+		Timeout:   common.HTTPTimeout,
+		Transport: http.DefaultTransport,
+	}
+	config.HTTPClient = httpc
+
+	if u.Scheme == common.TLSScheme {
```

Comment:
I'm a little uneasy about this - if the user has enabled TLS on the API interface it's to prevent information like credentials being observed as they go over the network. 

A bad actor that is in a position to observe plaintext/TLS traffic as it goes over the network is almost certainly in a position to MITM the traffic flow redirecting it to their machine where they present a bogus cert, which we would accept with `InsecureSkipVerify: true`, resulting in us sending the credentials to the bad actor anyway. Normally this is impossible because unless the bad actor had obtained a valid TLS certificate, the client would not establish the connection. I understand managing the TLS certs used by the operator is probably something we can do in the future, but we probably shouldn't work with TLS endpoints unless we're honouring the commitments to maintain the security it gives us in the meantime.  

The user could put the CA cert they're using for their API into their host bundle path - this would let the operator connect to the TLS endpoint _and_ do all the verification that is needed to make TLS secure without any additional code (the HTTP client loads the host's roots by default).

Reply:
Thanks for pointing that out. We can disable this for now and enable it when we are ready for it in the future.